### DISCUSSION TITLE
Menus containing actionable and selectable items

### DISCUSSION THREAD

**Discussion by jossmac** [adobe/react-spectrum#4762] [Ideas]
It's not uncommon for dropdown menus on the web to contain a mix of actionable and selectable items. Here's an example from Radix UI: https://www.radix-ui.com/docs/primitives/components/dropdown-menu

I think this could be partially achieved with some small changes to support `disabledBehavior` in `useMenu` and `useMenuItem`. Though it would be limited to single OR multiple selection within the menu, and of course items couldn't actually be disabled.

A better API might be to support `selectionMode` in `useMenuSection` but it's less clear what complexity that would introduce. Keen to know if this is something other folks are interested in.

**Comment by reidbarber** [comment]
This may be a duplicate of https://github.com/adobe/react-spectrum/issues/1621

**Reply by jossmac** [reply]
Yeah, that's the same intent. Though ideally the solution would be available to react-[aria|stately] consumers, not just Spectrum.

**Comment by jossmac** [comment]
Resolved by #7254

Related:
- [November 20, 2024 Release](https://react-spectrum.adobe.com/releases/2024-11-20.html)
- [Section-level selection docs](https://react-spectrum.adobe.com/react-aria/Menu.html#section-level-selection)
